settings/generic_setting.go

```diff
@@ -0,0 +1,280 @@
+		common.DataToStructPointer(d, resourceSchema, &setting)
+		var res string
+		switch defn := defn.(type) {
+		case workspaceSettingDefinition[T]:
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			res, err = retryOnEtagError[T, string](func(setting T) (string, error) { return defn.Update(ctx, w, setting) }, setting, defn.SetETag)
+			if err != nil {
+				return err
+			}
+		case accountSettingDefinition[T]:
+			a, err := c.AccountClient()
+			if err != nil {
+				return err
+			}
+			res, err = retryOnEtagError(func(setting T) (string, error) { return defn.Update(ctx, a, setting) }, setting, defn.SetETag)
+			if err != nil {
+				return err
+			}
+		default:
+			return fmt.Errorf("unexpected setting type: %T", defn)
+		}
+		d.SetId(res)
+		return nil
+	}
+
+	return common.Resource{
+		Schema: resourceSchema,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var setting T
+			return createOrUpdate(ctx, d, c, setting)
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var res *T
+			switch defn := defn.(type) {
+			case workspaceSettingDefinition[T]:
+				w, err := c.WorkspaceClient()
+				if err != nil {
+					return err
+				}
+				res, err = retryOnEtagError(func(etag string) (*T, error) { return defn.Read(ctx, w, etag) }, d.Id(), func(req *string, newEtag string) { *req = newEtag })
+				if err != nil {
+					return err
+				}
+			case accountSettingDefinition[T]:
+				a, err := c.AccountClient()
+				if err != nil {
+					return err
+				}
+				res, err = retryOnEtagError(func(etag string) (*T, error) { return defn.Read(ctx, a, etag) }, d.Id(), func(req *string, newEtag string) { *req = newEtag })
+				if err != nil {
+					return err
+				}
+			default:
+				return fmt.Errorf("unexpected setting type: %T", defn)
+			}
+			err := common.StructToData(res, resourceSchema, d)
+			if err != nil {
+				return err
+			}
+			// Update the etag. The server will accept any etag and respond
+			// with a response which is at least as recent as the etag.
+			// Updating, while not always necessary, ensures that the
+			// server responds with an updated response.
+			d.SetId(defn.GetETag(res))
+			return nil
+		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var setting T
+			defn.SetETag(&setting, d.Id())
+			return createOrUpdate(ctx, d, c, setting)
+		},
+		Delete: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var etag string
+			switch defn := defn.(type) {
+			case workspaceSettingDefinition[T]:
+				w, err := c.WorkspaceClient()
+				if err != nil {
+					return err
+				}
+				etag, err = retryOnEtagError(func(etag string) (string, error) { return defn.Delete(ctx, w, etag) }, d.Id(), func(req *string, newEtag string) { *req = newEtag })
```

Comment:
Do we really need to retry on the NotFound error? If it's not found then this typically it means that it was removed by external agent, so we just need to record deletion...

Reply:
Good point. Removed retry on `Not Found`, so we only retry on conflict.